### PR TITLE
[clkmgr] Fix documentation ambiguity around clock switching

### PR DESCRIPTION
**Pull request by tjaychen**
- Documentation fix specifically for clkmgr in #9273
- This PR is dependent upon the changes in #9347 

Signed-off-by: Timothy Chen <timothytim@google.com>

### CODE REVIEW

File: hw/ip/clkmgr/data/clkmgr.hjson.tpl [hunk 1]
```diff
@@ -187,11 +187,16 @@
           name: "SEL",
           mubi: true,
           desc: '''
-            A value of kMultiBitBool4True selects external clock as clock for the system.
-            While this register can always be programmed, it only takes effect when the system is in
-            life cycle TEST or RMA states when DFT is enabled.
+            When current value is not kMultiBitBool4True, a value of kMultiBitBool4True selects
+            external clock as clock for the system.  Any other value during this state has
+            no impact.
```
Comment (matutem) [inline hw/ip/clkmgr/data/clkmgr.hjson.tpl:192]:
```suggestion
            When the current value is not kMultiBitBool4True, writing a value of kMultiBitBool4True
            selects external clock as clock for the system, and writing any other value has no
            impact.
```

File: hw/ip/clkmgr/data/clkmgr.hjson.tpl [hunk 2]
```diff
@@ -187,11 +187,16 @@
...
 
-            All other values are invalid and keep clocks on internal sources.
+            When current value is kMultiBitBool4True, a value of kMultiBitBool4False selects
+            internal clock as clock for the system. Any other value during this stage has no
+            impact.
```
Comment (matutem) [inline hw/ip/clkmgr/data/clkmgr.hjson.tpl:196]:
```suggestion
            When the current value is kMultiBitBool4True, writing a value of kMultiBitBool4False
            selects internal clock as clock for the system, and writing any other value has no
            impact.
```

File: hw/ip/clkmgr/data/clkmgr.hjson.tpl [hunk 3]
```diff
@@ -187,11 +187,16 @@
...
+
+            While this register can always be programmed, it only takes effect when the system
+            is in life cycle TEST, DEV or RMA states when hardware debug is enabled.
```
Comment (matutem) [inline hw/ip/clkmgr/data/clkmgr.hjson.tpl:199]: https://docs.opentitan.org/doc/security/specs/device_life_cycle/ seems to say dft is not enabled in DEV state, so

```suggestion

            While this register can always be programmed, it only takes effect when DFT functions
            are enabled, which is in life cycle TEST_UNLOCKED or RMA states.
```

**Review by matutem — CHANGES_REQUESTED** [review]

**Comment by tjaychen** [comment]
updated! please have a look.

**Review by matutem — APPROVED** [review]
LGTM

**Pull request merged**